Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs

```diff
@@ -200,6 +204,57 @@ public List<string> GetPlatforms(string productName, string version)
 
             return platforms.OrderBy(p => p).ToList();
         }
+        
+        public BackupInformationModel RunBackup()
+        {
+            var timeStamp = DateTime.UtcNow.ToString("yyyy-MM-dd_HH-mm-ss");
+            
+            string backupFileName = "backup_" + timeStamp + ".zip";
+            string backupArchiveFileName = Path.Combine(BackupRoot, backupFileName); 
+            
+            //Clear the backup folder first
+            var backupDirectoryInfo = new DirectoryInfo(BackupRoot);
+
+            if (backupDirectoryInfo.Exists)
+            {
+                backupDirectoryInfo.DeleteContent();
+            }
+            else
+            {
+                backupDirectoryInfo.Create();
+            }
+            
+            //Create the backup -> zip the whole ArtifactRoot folder
+            ZipFile.CreateFromDirectory(ArtifactRoot, backupArchiveFileName);
+
+            return new BackupInformationModel
+            {
+                FullPath = backupArchiveFileName,
+                FileName = backupFileName
+            };
+        }
+
+        public void RestoreBackup(ZipArchive backupPayload)
+        {
+            try
+            {
+                //Clear the whole artifact root directory or create it, if it's not existing
+                if (ArtifactRootDir.Exists) 
+                {
+                    ArtifactRootDir.DeleteContent();  
+                }
+                else {
+                    ArtifactRootDir.Create();
+                }
+                
+                backupPayload.ExtractToDirectory(ArtifactRoot);
+            }
+            catch (Exception e)
+            {
+                Logger.LogCritical("unexpected error during restoring the backup");
```

Comment:
Please log the exception message in order to provide a more specific error message.

Reply:
Realized, see:

https://github.com/Traeger-GmbH/release-server/blob/feature/implement-backup-mechanism/Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs#L252-L255